app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt

```diff
@@ -257,8 +257,8 @@ class BrowserFragment : Fragment(), BackHandler {
             )
         }
         toolbarComponent.getView().setOnSiteSecurityClickedListener {
-            sessionId?.run {
-                val session = requireNotNull(requireContext().components.core.sessionManager.findSessionById(this))
+            val session = requireContext().components.core.sessionManager.selectedSession
+            session?.run {
```

Comment:
This probably fixes it in normal sessions, but breaks it in custom tabs (where the session id is fixed) - or do we not want to show that in custom tabs?

Reply:
I guess even if we not want to show it: Right now it may show the doorhanger for the selected session in the custom tab?  :)

Reply:
You are right, now its breaking custom tabs. For some reasons I'm getting a `null` `sessionId` after the user returns to the same tab but by entering the `url`  or by selecting it from clipboard in the `HomeFragment`.

I will try to find a solution that work for both.

Reply:
To be honest I didn't know that the `doorhanger` was working on custom tabs XD

Reply:
An alternative could be to use the `sessionId` as a primary option and if it is `null` fallback to use the currently selected tab. @pocmo  What do you think?

Reply:
@colintheshots any ideas why the `sessionId` is `null`?

Reply:
I noticed  the [FullScreenFeature is using the same approuch as I described before](https://github.com/mozilla-mobile/android-components/blob/c9a2d3d8c7967a577b59e94f29031b3421b59855/components/feature/session/src/main/java/mozilla/components/feature/session/FullScreenFeature.kt#L29). If `sessionId` is `null` is using the `selected session `. I'll go with the solution if you have any suggestion please let me know.

Reply:
Sorry, we should've documented that decision. We needed to pass an argument into BrowserFragment when there is an existing session already opened, but it was valid to have none. An elvis operator wouldn't be sufficient as we needed to handle the null case differently.